src/main/java/com/danielagapov/spawn/Services/User/UserService.java

```diff
@@ -39,12 +46,12 @@ public UserDTO getUserById(UUID id) {
     }
 
     public List<UserDTO> getUsersByTagId(UUID tagId) {
-        // TODO: change this logic later, once tags are setup.
-        try {
-            return UserMapper.toDTOList(repository.findAll());
-        } catch (DataAccessException e) {
-            throw new RuntimeException("Error retrieving users", e);
+        List<UserFriendTagMapping> mappings = userTagMappingRepository.findByFriendTagId(tagId);
```

Comment:
this could throw exceptions

Reply:
So, I don't think it's a problem if there aren't any found -> we don't need to catch something like `EntityNotFoundException`, but, we should surround it with a try-catch with these, probably:

```
} catch (DataAccessException e) {
        // Handle database-related issues
        throw new DatabaseException("Error accessing data for tag ID: " + tagId, e);
    } catch (Exception e) {
        // Handle unexpected issues
        throw new ApplicationException("An unexpected error occurred while fetching users by tag ID.", e);
    }
```